### DISCUSSION TITLE
Sign arbitrary data file and produce detached PKCS#7 signature file?

### DISCUSSION THREAD

**Discussion by wilx** [ebourg/jsign#158] [Q&A] [answered]
Is it possible to use this utility to sign arbitrary data file and produce PKCS#7 detached signature which would be verifiable via, e.g., [`signver`](https://manpages.ubuntu.com/manpages/jammy/man1/signver.1.html) utility?

**Comment by ebourg** [comment]
Jsign can generate a detached PKCS#7 signature, but I don't think signver will be able to verify it. As I understand signver hashes the whole file, but an Authenticode signature hash may skip some parts of the file (such as the checksum field in a PE file, or the header in a MSI file).

**Comment by wilx** [comment]
Signing an entire file is what I want.

**Comment by ebourg** [comment, marked as answer]
Another point worth noting, the object signed by Authenticode is a SpcIndirectDataContent structure containing the hash of the file. Unless signver knows how to build this structure it won't be able to verify the signature.
